feat: migrate from npm to bun package manager

DBosley opened the pull request:
## Summary

- Migrates the project from npm to bun package manager (v1.3.2)
- Updates all GitHub Actions workflows to use bun
- Updates all documentation and scripts to reference bun instead of npm
- Simplifies E2E test workflow by removing Docker container dependency

## Changes

### Package Manager Migration
- `package.json`: Scripts updated to use bun
- `netlify.toml`: Added `BUN_VERSION = "1.3.2"` and updated build commands
- Added empty `bun.lockb` for Netlify auto-detection (Netlify only detects binary lockfile, not text `bun.lock`)

### GitHub Workflows
All workflows updated to use `oven-sh/setup-bun@v2` with pinned version `1.3.2`:
- `build.yml`
- `code-quality.yml`
- `publish.yml`
- `test.yml`

### E2E Test Workflow Simplification
Removed the Playwright Docker container (`mcr.microsoft.com/playwright:v1.54.1-noble`) in favor of Playwright's recommended CI setup:

**Before:**
```yaml
container:
  image: mcr.microsoft.com/playwright:v1.54.1-noble
  options: --user 1001
```

**After:**
```yaml
- name: Install Playwright browsers
  run: bunx playwright install --with-deps
```

This follows [Playwright's official CI documentation](https://playwright.dev/docs/ci) which recommends using `playwright install --with-deps` on `ubuntu-latest` instead of Docker containers. The `--with-deps` flag automatically installs all required browser system dependencies.

Benefits:
- Simpler configuration
- No Docker image version synchronization issues
- Native bun/node tooling support
- Faster startup (no container overhead)

### Documentation Updates
- `README.md`, `CONTRIBUTING.md`, `CLAUDE.md`
- `tests/e2e/README.md`, `E2E_TESTS.md`
- Shell scripts: `update-sdk.sh`, `lint.sh`, `pre-commit.sh`, `pre-commit-check.sh`

## Test Plan
- [x] All GitHub Actions workflows pass (ESLint, Prettier, Unit Tests, Build, E2E)
- [x] Netlify deployments work with bun
- [x] Local development verified with `bun install` and `bun run dev`

DBosley committed:
feat: add bun package manager support

- Add bun.lock and bunfig.toml for bun package manager
- Simplify overrides to global format (bun doesn't support nested overrides)
- Add 24-hour minimum release age security policy with wormhole package exclusions
- Verified all 100 dependencies match npm lockfile versions exactly

The package-lock.json is retained for npm fallback compatibility.

priyajeet commented on .github/workflows/build.yml:
```diff
@@ -35,12 +38,14 @@ jobs:
+      - uses: oven-sh/setup-bun@v2
+        if: steps.filter.outputs.tokens == 'true' || github.ref == 'refs/heads/development'
+        with:
+          bun-version: '1.3.2'
```
1.3.4 is latest

DBosley replied:
fixed

priyajeet commented on tests/e2e/README.md:
````diff
@@ -16,13 +16,13 @@ Before running the tests, ensure you have the following installed:
    Make sure you have clean installed all dependencies:
 
    ```sh
-   npm ci
````
btw, bun ci also works

DBosley replied:
Good call, I'll swap that

DBosley committed:
chore: complete bun migration with workflow updates

- Update all GitHub workflows to use bun with oven-sh/setup-bun@v2
- Pin bun version to 1.3.2 across all workflows
- Simplify E2E tests by removing Playwright container
- Use native playwright install --with-deps instead
- Update documentation and scripts for bun usage
- Add netlify.toml bun configuration

priyajeet commented on .github/workflows/publish.yml:
```diff
@@ -17,22 +17,23 @@ jobs:
-          node-version: 20
-          cache: 'npm'
-          cache-dependency-path: '**/package-lock.json'
+          node-version: 22
```
24 is the new lts

DBosley replied:
This was an inconsistency with the rest of the node versions. I want to update to 24 in the next PR
